snapcraft/internal/errors.py

```diff
@@ -454,6 +454,18 @@ def __init__(self, *, elf_file, process_exception, patchelf_version):
                          patchelf_version=patchelf_version)
 
 
+class StagePackageMissingError(SnapcraftError):
+
+    fmt = (
+        '{package!r} is required as a `stage-packages` entry for this part to '
+        'work properly.\n'
+        'Add {package!r} as a `stage-packages` entry for this part.'
```

Comment:
This doesn't flow super well, it feels like it's being shoved into our new error structure instead of massaging it :stuck_out_tongue: .

How about:

```
'{package!r} is required inside the snap for this part to work properly.\n'
'Add it as a `stage-packages` entry for this part.'
```

Reply:
lol, ok, I just wished you raised this earlier as the PR is finally green ;-)

Reply:
is this the last tidbit on your side?